networkx/algorithms/shortest_paths/dense.py

```diff
@@ -13,13 +13,22 @@
 def floyd_warshall_numpy(G, nodelist=None, weight="weight"):
     """Find all-pairs shortest path lengths using Floyd's algorithm.
 
+    This algorithm for finding shortest paths takes advantage of
+    matrix representations of a graph and works well for dense
+    graphs where all-pairs shortest path is desired. The results
+    are returns in a numpy array with each column and row
+    representing a node and entries providing the distance along
+    the shortest path between that row's node and column's node.
```

Comment:
Some commas or colons would help read this more easily.

I could also see a slightly different phrasing working, but the above should be fine with some punctuation / splitting into multiple sentences.

> The results are returned as a NumPy array, ``distance[i, j]``, where ``i`` and ``j`` represent nodes.  The entry ``distance[i, j]`` is the distance along the shortest path between ``i`` and ``j``.

Reply:
Thanks for these suggestions! It reads much better.